pywren/future.py

```diff
@@ -103,10 +100,10 @@ def result(self, timeout=None, check_only=False, throw_except=True):
             else:
                 return None
 
+        if storage_handler is None:
```

Comment:
Did we reach consensus on what the model we want to support is ? There was the other option of passing in the storage_handler while constructing the future that I think avoids some of the failure scenarios ?